### ISSUE TITLE
Fix S3254: False positive when used in Expression

### ISSUE THREAD

**Issue by ohadschn** [SonarSource/sonar-csharp#789]
### Description

Default parmaeters cannot be used inside expression trees (_CS0854: An expression tree may not contain a call or invocation that uses optional arguments_), so the advice given by S3254 cannot be followed in those cases.

### Repro steps

```cs
private static string FuncWithOptionals(string str = null)  
{ 
    return str;
}

public static void TestS3254()
{
    Expression<Action> e = () => FuncWithOptionals(null); //S3254 (if null is removed, CS0854)
}
```
### Related information

* SonarC# Version: **6.4.0.3347**
* Visual Studio Version: **2017 Enterprise 15.3.4**

**Comment by valhristov** [comment]
Nice catch! Thanks for the feedback.
